src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -25,10 +25,264 @@
+    pydmlStmt = pydmlStmt + [getNamedArgOrEmpty('check', check), getNamedArgOrEmpty('par', par), getNamedArgOrEmpty('mode', mode), getNamedArgOrEmpty('taskpartitioner', taskpartitioner), getNamedArgOrEmpty('tasksize', tasksize), getNamedArgOrEmpty('datapartitioner', datapartitioner), getNamedArgOrEmpty('resultmerge', resultmerge), getNamedArgOrEmpty('opt', opt) ]
+    pydmlStmt = pydmlStmt + ['):\n']
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt))
+    MLContext.globalMLContext.numTabs += 1
+    yield 
+    MLContext.globalMLContext.numTabs -= 1
+    pydmlStmt = [getTabs(), '\n']
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt))
+
+@contextmanager
+def FOR(s, start, end):
+    scopeCheck('FOR')
+    pydmlStmt = [getTabs(), 'for(', s.getID(), ' in ', getPythonFloatValOrScalarID(start), ':', getPythonFloatValOrScalarID(end), '):\n']
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt))
+    MLContext.globalMLContext.numTabs += 1
+    yield
+    MLContext.globalMLContext.numTabs -= 1
+    pydmlStmt = [getTabs(), '\n']
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt))
+
+def getPythonFloatValOrScalarID(value):
+    if isinstance(value, float) or isinstance(value, int) or isinstance(value, long):
+        return str(float(value))
+    elif isinstance(value, Scalar):
+        return value.getID()
+    else:
+        raise Exception('Incorrect type: expected float or int or long or Scalar')
+        
+def scopeCheck(fnName):
+    if MLContext.globalMLContext is None:
+        raise Exception('Incorrect usage: \"' + fnName + '\" can only run if MLContext.start has been called')
 
+class random(object):
+    @staticmethod
+    def normal(rows=None, cols=None, sparsity=None):
+        if rows is None or cols is None:
+            raise Exception('rows and cols are mandatory arguments')
+        return simpleAddPyDMLStmt('random.normal', args=['rows=', str(rows), ', cols=', str(cols), ', sparsity=', getDefault(sparsity, 1.0)])
+    @staticmethod
+    def uniform(rows=None, cols=None, sparsity=None, min=None, max=None):
+        if rows is None or cols is None:
+            raise Exception('rows and cols are mandatory arguments')
+        return simpleAddPyDMLStmt('random.uniform', args=['rows=', str(rows), ', cols=', str(cols), ', sparsity=', getDefault(sparsity, 1.0), ", min=", getDefault(min, 0.0), ", max=", getDefault(max, 1.0) ])
+
+def getDefault(val, defaultVal):
+    if val is None:
+        return str(defaultVal)
+    else:
+        return val
+
+def getNamedArgOrEmpty(name, val):
+    if val is None:
+        return ''
+    else:
+        return ', ' + name + '=' + str(val)
+
+class Matrix(object):
+    systemmlVarID = 0
+    def __init__(self, value):
+        if isinstance(value, DataFrame):
+            self.df = df
+        Matrix.systemmlVarID += 1
+        self.ID = 'mVar' + str(Matrix.systemmlVarID)
+    def getID(self):
```

Comment:
stylistic comment: Can you space the methods out with a space in between, so that it is easier to read?